bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -294,28 +309,73 @@ protected static Credentials extractCredentials(String url) {
 
     /**
      * Factory method to create a {@link HttpMethod}-object according to the
-     * given String <code>httpMethod</code>
+     * given String <code>httpMethodString</code>
      *
      * @param httpMethodString the name of the {@link HttpMethod} to create
-     * @param url
      *
-     * @return an object of type {@link GetMethod}, {@link PutMethod}, {@link PostMethod} or {@link DeleteMethod}
      * @throws IllegalArgumentException if <code>httpMethod</code> is none of <code>GET</code>, <code>PUT</code>,
      *             <code>POST</POST> or <code>DELETE</code>
      */
-    public static HttpMethod createHttpMethod(String httpMethodString, String url) {
+    public static HttpMethod createHttpMethod(String httpMethodString) {
 
         if ("GET".equals(httpMethodString)) {
-            return new GetMethod(url);
+            return HttpMethod.GET;
         } else if ("PUT".equals(httpMethodString)) {
-            return new PutMethod(url);
+            return HttpMethod.PUT;
         } else if ("POST".equals(httpMethodString)) {
-            return new PostMethod(url);
+            return HttpMethod.POST;
         } else if ("DELETE".equals(httpMethodString)) {
-            return new DeleteMethod(url);
+            return HttpMethod.DELETE;
         } else {
             throw new IllegalArgumentException("given httpMethod '" + httpMethodString + "' is unknown");
         }
     }
 
+    /**
+     * Creates authentication result for basic authentication.
+     * When this result is added to the {@link AuthenticationStore},
+     * the {@link HttpClient} will send authentication header preemptively.
+     *
+     * @param host - the hostname
+     * @param port - the port, if set to not positive number it will be ignored
+     * @param user - the name of the user that is requiring the resource
+     * @param password - the user password
+     * @param isProxyAuthentication - parameter that specifies, if the result will be used for proxy authenticaiton
+     */
+    public static Authentication.Result createBasicAuthenticationResult(String host, int port, String user,
```

Comment:
Besides this, you are making Authentication.Result part of the API of HttpUtil, which is not good (it would make a refactoring like the one in this PR impossible).